src/hotspot/share/opto/superword.hpp

```diff
@@ -563,11 +561,35 @@ class SuperWord : public ResourceObj {
   bool SLP_extract();
 
   // Find the "seed" memops pairs. These are pairs that we strongly suspect would lead to vectorization.
+  class MemOp : public StackObj {
+  private:
+    MemNode* _mem;
+    const VPointer* _vpointer;
+    int _original_index;
+
+  public:
+    // Empty, for GrowableArray
+    MemOp() :
+      _mem(nullptr),
+      _vpointer(nullptr),
+      _original_index(-1) {}
+    MemOp(MemNode* mem, const VPointer* vpointer, int original_index) :
+      _mem(mem),
+      _vpointer(vpointer),
+      _original_index(original_index) {}
+
+    MemNode* mem() const { return _mem; }
+    const VPointer& vpointer() const { return *_vpointer; }
+    int original_index() const { return _original_index; }
+
+    static int cmp_by_group(MemOp* a, MemOp* b);
+    static int cmp_by_group_and_con_and_original_index(MemOp* a, MemOp* b);
+  };
```

Comment:
Note: we used to put `VPointer` directly in a `GrowableArray` to sort them. Now that the `VPointer` does not have a reference to the `MemNode*`, we have to create a "tuple" of `(VPointer, mem)`. The `original_index` is needed to keep the sort stable in `qsort`. Before this was done with the `mem->_idx`, which is a bit less well behaved.